tests/ert/ui_tests/cli/analysis/test_design_matrix.py

```diff
@@ -300,3 +301,97 @@ def _evaluate(coeffs, x):
+        pd.DataFrame([]),
+    )
+
+    with open("poly.ert", "w", encoding="utf-8") as f:
+        f.write(
+            dedent(
+                """\
+                QUEUE_OPTION LOCAL MAX_RUNNING 50
+                RUNPATH poly_out/realization-<IENS>/iter-<ITER>
+                OBS_CONFIG observations
+                NUM_REALIZATIONS 10
+                GEN_KW COEFFS_A coeff_priors_a
+                GEN_KW COEFFS_B coeff_priors_b
+                GEN_KW COEFFS_C coeff_priors_c
+                GEN_DATA POLY_RES RESULT_FILE:poly.out
+                DESIGN_MATRIX design_matrix.xlsx DESIGN_SHEET:DesignSheet01 DEFAULT_SHEET:DefaultSheet
+                INSTALL_JOB poly_eval POLY_EVAL
+                FORWARD_MODEL poly_eval
+                """
+            )
+        )
+
+    with open("poly_eval.py", "w", encoding="utf-8") as f:
+        f.write(
+            dedent(
+                """\
+                #!/usr/bin/env python3
+                import json
+
+                def _load_coeffs(filename):
+                    with open(filename, encoding="utf-8") as f:
+                        params = json.load(f)
+                        params = params["COEFFS_A"] | params["COEFFS_B"] | params["COEFFS_C"]
+                        return params
+
+                def _evaluate(coeffs, x):
+                    return coeffs["a"] * x**2 + coeffs["b"] * x + coeffs["c"]
+
+                if __name__ == "__main__":
+                    coeffs = _load_coeffs("parameters.json")
+                    output = [_evaluate(coeffs, x) for x in range(10)]
+                    with open("poly.out", "w", encoding="utf-8") as f:
+                        f.write("\\n".join(map(str, output)))
+                """
+            )
+        )
+    os.chmod(
+        "poly_eval.py",
+        os.stat("poly_eval.py").st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH,
+    )
+
+    with open("coeff_priors_a", "w", encoding="utf-8") as f:
+        f.write("a UNIFORM 0 1")
+    with open("coeff_priors_b", "w", encoding="utf-8") as f:
+        f.write("b UNIFORM 0 2")
+    with open("coeff_priors_c", "w", encoding="utf-8") as f:
+        f.write("c UNIFORM 0 5")
+
+    run_cli(
+        ES_MDA_MODE,
+        "--disable-monitoring",
+        "poly.ert",
+        "--experiment-name",
+        "test-experiment",
+    )
+    storage_path = ErtConfig.from_file("poly.ert").ens_path
+    coeffs_a_previous = None
+    with open_storage(storage_path) as storage:
+        experiment = storage.get_experiment_by_name("test-experiment")
+        for i in range(4):
+            ensemble = experiment.get_ensemble_by_name(f"default_{i}")
+
+            coeffs_a = ensemble.load_parameters("COEFFS_A")["values"].values.flatten()
+            if coeffs_a_previous is not None:
+                assert not np.array_equal(coeffs_a, coeffs_a_previous)
+            coeffs_a_previous = coeffs_a
+
+            coeffs_b = ensemble.load_parameters("COEFFS_B")["values"].values.flatten()
```

Comment:
Just wondering what to expect here. Currently `DESIGN_MATRIX:b` will adopt new group name `COEFFS_B` (which could be part of the test), but should this be the same for all iterations?

Reply:
It was agreed to keep as is In the current version at least.